packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -418,6 +418,9 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
     const rowBuffer = !disableVirtualization ? rootProps.rowBuffer : 0;
     const columnBuffer = !disableVirtualization ? rootProps.columnBuffer : 0;
 
+    let isFocusedCellRendered = false;
+    const { cell } = apiRef.current.state.focus;
```

Comment:
```suggestion
    const focusedCell = gridFocusCellSelector(apiRef);
```
Use the selector instead